pkg/provisioner/provisioner.go

```diff
@@ -53,7 +57,7 @@ type Provisioner interface {
 	// Provision writes the image from the host spec to the host. It
 	// may be called multiple times, and should return true for its
 	// dirty flag until the deprovisioning operation is completed.
-	Provision(getUserData UserDataSource) (result Result, err error)
+	Provision(getUserData UserDataSource, hardwareConfig HardwareConfig) (result Result, err error)
```

Comment:
The provisioner has a copy of the entire host in its "host" property. It should not need any other information to decide how to configure the host, so we should not need to add an argument to the Provision() method like this.

Reply:
Done. Removed this argument